aten/src/ATen/cpu/vec256/vec256_bfloat16.h

```diff
@@ -186,13 +186,109 @@ template <> class Vec256<BFloat16> {
     auto o2 = vop(hi);
     return cvtfp32_bf16(o1, o2);
   }
+  Vec256<BFloat16> abs() const {
+    __m256 lo, hi;
+    cvtbf16_fp32(values, lo, hi);
+    auto mask = _mm256_set1_ps(-0.f);
+    auto o1 = _mm256_andnot_ps(mask, lo);
+    auto o2 = _mm256_andnot_ps(mask, hi);
+    return cvtfp32_bf16(o1, o2);
+  }
+  Vec256<BFloat16> angle() const {
+    return _mm256_set1_epi16(0);
+  }
+  Vec256<BFloat16> real() const {
+    return *this;
+  }
+  Vec256<BFloat16> imag() const {
+    return _mm256_set1_epi16(0);
+  }
+  Vec256<BFloat16> conj() const {
+    return *this;
+  }
   Vec256<BFloat16> exp() const {
     return map(Sleef_expf8_u10);
   }
   Vec256<BFloat16> log() const {
     return map(Sleef_logf8_u10);
   }
-
+  Vec256<BFloat16> frac() const;
+  Vec256<BFloat16> floor() const {
+    __m256 lo, hi;
+    cvtbf16_fp32(values, lo, hi);
+    auto o1 = _mm256_floor_ps(lo);
+    auto o2 = _mm256_floor_ps(hi);
+    return cvtfp32_bf16(o1, o2);
+  }
+  Vec256<BFloat16> neg() const {
+    __m256 lo, hi;
+    cvtbf16_fp32(values, lo, hi);
+    auto mask = _mm256_set1_ps(-0.f);
+    auto o1 = _mm256_xor_ps(mask, lo);
+    auto o2 = _mm256_xor_ps(mask, hi);
+    return cvtfp32_bf16(o1, o2);
+  }
+  Vec256<BFloat16> trunc() const {
+    __m256 lo, hi;
+    cvtbf16_fp32(values, lo, hi);
+    auto o1 = _mm256_round_ps(lo, (_MM_FROUND_TO_ZERO | _MM_FROUND_NO_EXC));
+    auto o2 = _mm256_round_ps(hi, (_MM_FROUND_TO_ZERO | _MM_FROUND_NO_EXC));
+    return cvtfp32_bf16(o1, o2);
+  }
+  Vec256<BFloat16> sqrt() const {
+    __m256 lo, hi;
+    cvtbf16_fp32(values, lo, hi);
+    auto o1 = _mm256_sqrt_ps(lo);
+    auto o2 = _mm256_sqrt_ps(hi);
+    return cvtfp32_bf16(o1, o2);
+  }
+  Vec256<BFloat16> reciprocal() const {
+    __m256 lo, hi;
+    cvtbf16_fp32(values, lo, hi);
+    auto ones = _mm256_set1_ps(1);
+    auto o1 = _mm256_div_ps(ones, lo);
+    auto o2 = _mm256_div_ps(ones, hi);
+    return cvtfp32_bf16(o1, o2);
+  }
+  Vec256<BFloat16> rsqrt() const {
+    __m256 lo, hi;
+    cvtbf16_fp32(values, lo, hi);
+    auto ones = _mm256_set1_ps(1);
+    auto o1 = _mm256_div_ps(ones, _mm256_sqrt_ps(lo));
+    auto o2 = _mm256_div_ps(ones, _mm256_sqrt_ps(hi));
+    return cvtfp32_bf16(o1, o2);
+  }
+  // Comparison using the _CMP_**_OQ predicate.
+  //   `O`: get false if an operand is NaN
+  //   `Q`: do not raise if an operand is NaN
+  template<int imm8>
+  Vec256<BFloat16> compare(const Vec256<BFloat16>& other) const {
```

Comment:
Would you mind letting #35844 and #36106 implement the comparison operators? The reason is that those two PRs reduce code redundancy by using the same block of code across comparison, bitwise, and arithmetic operators. This function here virtually repeated the code snippet that implements other binary operators. Thanks!

Reply:
@xuhdev , there has no conflict with them,  we can apply your PRs after @ngimel merge this PR. Thanks!